### PR TITLE
CDAP-5045: Timeout when enabling explore on app deploy

### PR DESCRIPTION
**Pull request by jwang47**
https://issues.cask.co/browse/CDAP-5045
http://builds.cask.co/browse/CDAP-RBT645

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java [hunk 1]
```diff
@@ -69,6 +70,7 @@
   private static final Gson GSON = new Gson();
   private static final Type SUMMARY_LIST_TYPE = new TypeToken<List<DatasetSpecificationSummary>>() { }.getType();
   private static final Type MODULE_META_LIST_TYPE = new TypeToken<List<DatasetModuleMeta>>() { }.getType();
+  private static final HttpRequestConfig REQUEST_CONFIG = new HttpRequestConfig(60000, 60000);
```
Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: we want this to be configurable.

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: I think if this is above any "reasonable" value, then it need not be configurable. We tested on HDInsights, and 30 was sufficient. So we're making it 60 to be on the safe side.

If someone's network is so much slower than that, I don't think we can expect the overall system to work anyways.
@dereklwood Any input?

Comment (dereklwood) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: +1 for configurable.  IMO we shouldn't make any assumptions about any environment, and doesn't hurt anything to have a configuration option

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: Alright. Suggestions for configuration name?
`http.client.read.timeout`
`http.client.write.timeout`
?

Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: ``http.client.connection.timeout``?

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: ok yea, I prefer that also. (1 better than 2)

File: cdap-client-tests/src/test/java/co/cask/cdap/client/QueryClientTestRun.java [hunk 2]
```diff
@@ -55,6 +56,8 @@
   @Before
   public void setUp() throws Throwable {
     super.setUp();
+
+    CConfiguration cConf = CConfiguration.create();
```
Comment (anwar6953) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/QueryClientTestRun.java]: unused

File: cdap-client/src/main/java/co/cask/cdap/client/QueryClient.java [hunk 3]
```diff
@@ -75,7 +76,12 @@ public Boolean get() {
       }
     };
 
-    exploreClient = new SuppliedAddressExploreClient(hostname, port, accessToken, sslEnabled, verifySSLCert);
+    exploreClient = new SuppliedAddressExploreClient(hostname, port, accessToken, sslEnabled, verifySSLCert) {
```
Comment (anwar6953) [inline cdap-client/src/main/java/co/cask/cdap/client/QueryClient.java]: This class is not used internally ever?

Comment (jwang47) [inline cdap-client/src/main/java/co/cask/cdap/client/QueryClient.java]: Which class?

Comment (anwar6953) [inline cdap-client/src/main/java/co/cask/cdap/client/QueryClient.java]: QueryClient

Comment (jwang47) [inline cdap-client/src/main/java/co/cask/cdap/client/QueryClient.java]: No, QueryClient isn't used internally.

Comment (anwar6953) [inline cdap-client/src/main/java/co/cask/cdap/client/QueryClient.java]: ok. that's fine, then.

**Comment by anwar6953** [comment]
LGTM

**Pull request merged**